src/chat.rs

```diff
@@ -3135,22 +3135,13 @@ pub async fn send_text_msg(
 
 /// Sends chat members a request to edit the given message's text.
 pub async fn send_edit_request(context: &Context, msg_id: MsgId, new_text: String) -> Result<()> {
-    let mut original_msg = Message::load_from_db(context, msg_id).await?;
-    ensure!(
-        original_msg.from_id == ContactId::SELF,
-        "Can edit only own messages"
-    );
-    ensure!(!original_msg.is_info(), "Cannot edit info messages");
-    ensure!(!original_msg.has_html(), "Cannot edit HTML messages");
     ensure!(
-        original_msg.viewtype != Viewtype::VideochatInvitation,
-        "Cannot edit videochat invitations"
-    );
-    ensure!(
-        !original_msg.text.is_empty(), // avoid complexity in UI element changes. focus is typos and rewordings
-        "Cannot add text"
+        message::can_send_edit_request(context, msg_id).await?,
```

Comment:
This leads to double-checking of `why_cant_send_ex()` which is expensive (another call is done by `prepare_send_msg()`), so i suggest to only leave additional checks here (and only if they aren't done later). `can_send_edit_request()` looks rather as a function for UIs, not for internal usage.

Reply:
i do not think, that the call is _so_ expensive, it would be called one time when the pop up menu is shown, and another time before the message is sent. if it really turns out that this slows down things significantly, we can think over - otherwise, i would prefer this additional protection against things going wrong in UI

Reply:
It will be called the third time from `send_msg()` which is called from this function, i think this is too much. It calls `is_self_in_chat()` and `check_securejoin_wait()`, so it makes several queries to the db. Otoh editing messages isn't a frequent operation, so maybe it's not critical